File: Programmers/타겟넘버.py
#dfs
def solution(numbers, target):
    answer = []
    def dfs(index,target,sum):
        if index==len(numbers):
            if target==sum:
                answer.append(1)
                return
            else:
                return
        else:
            dfs(index+1,target,sum-numbers[index])
            dfs(index+1,target,sum+numbers[index])
    dfs(0,target,0)
    answer=len(answer)
    return answer

File: Programmers/타겟넘버.js
function solution(numbers, target) {
  let answer = 0;
  let v = 0;
  let index = 0;
  function dfs(v, index) {
    if (index === numbers.length) {
      if (v === target) {
        answer += 1;
      }
    } else {
      dfs(v - numbers[index], index + 1);
      dfs(v + numbers[index], index + 1);
    }
  }
  dfs(0, index);
  return answer;
}
